From: Simon McVittie <smcv@debian.org>
Date: Sat, 16 Feb 2019 16:19:30 +0000
Subject: dconf: Create a temporary XDG_RUNTIME_DIR

Some chroot-based autobuilder environments, including Debian's sbuild,
have a tendency to inherit the XDG_RUNTIME_DIR variable from outside
the chroot but not create a corresponding directory inside the chroot.

If we do have a valid XDG_RUNTIME_DIR, we probably don't want to
touch it during the build anyway: it's better to use a temporary one
in all cases.

The upstream is going to remove the dconf stuff so there's no
need to upstream. See https://github.com/ibus/ibus/issues/2150

Forwarded: not-needed
Reviewed-by: Simon McVittie <smcv@debian.org>
Last-Update: 2019-12-27
---
 data/dconf/make-dconf-override-db.sh | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/data/dconf/make-dconf-override-db.sh b/data/dconf/make-dconf-override-db.sh
index dddbcec..fc85813 100755
--- a/data/dconf/make-dconf-override-db.sh
+++ b/data/dconf/make-dconf-override-db.sh
@@ -5,8 +5,9 @@ set -e
 export TMPDIR=$(mktemp -d --tmpdir="$PWD")
 export XDG_CONFIG_HOME="$TMPDIR/config"
 export XDG_CACHE_HOME="$TMPDIR/cache"
+export XDG_RUNTIME_DIR="$TMPDIR/run"
 export GSETTINGS_SCHEMA_DIR="$TMPDIR/schemas"
-mkdir -p $XDG_CONFIG_HOME $XDG_CACHE_HOME $GSETTINGS_SCHEMA_DIR
+mkdir -p $XDG_CONFIG_HOME $XDG_CACHE_HOME $GSETTINGS_SCHEMA_DIR $XDG_RUNTIME_DIR
 
 trap 'rm -rf $TMPDIR' ERR
 
